plugins/linuxv2/ifplugin/descriptor/interface_veth.go

```diff
@@ -0,0 +1,194 @@
+		return nil, err
+	}
+
+	// move to the namespace with the interface
+	revert, err := intfd.nsPlugin.SwitchToNamespace(nsCtx, linuxIf.Namespace)
+	if err != nil {
+		intfd.log.Error(err)
+		return nil, err
+	}
+	defer revert()
+
+	// rename from the temporary host name to the requested host name
+	intfd.ifHandler.RenameInterface(tempHostName, hostName)
+	if err != nil {
+		intfd.log.Error(err)
+		return nil, err
+	}
+
+	// build metadata
+	link, err := intfd.ifHandler.GetLinkByName(hostName)
+	if err != nil {
+		intfd.log.Error(err)
+		return nil, err
+	}
+	metadata = &ifaceidx.LinuxIfMetadata{
+		Namespace:    linuxIf.Namespace,
+		LinuxIfIndex: link.Attrs().Index,
+	}
+
+	return metadata, nil
+}
+
+// deleteVETH either un-configures one VETH-end if the other end is still configured, or
+// removes the entire VETH pair.
+func (intfd *InterfaceDescriptor) deleteVETH(nsCtx nslinuxcalls.NamespaceMgmtCtx, key string, linuxIf *interfaces.LinuxInterface, metadata *ifaceidx.LinuxIfMetadata) error {
+	// determine host/logical/temporary interface names
+	hostName := getHostIfName(linuxIf)
+	peerName := getVethPeerName(linuxIf)
+	tempHostName := getVethTemporaryHostName(linuxIf.Name)
+	tempPeerHostName := getVethTemporaryHostName(peerName)
+
+	// check if the other end is still configured
+	ifIndex := intfd.scheduler.GetMetadataMap(InterfaceDescriptorName)
+	_, peerExists := ifIndex.GetValue(peerName)
+	if peerExists {
+		// just un-configure this VETH-end, but do not delete the pair
+
+		// rename to the temporary host name
+		err := intfd.ifHandler.RenameInterface(hostName, tempHostName)
+		if err != nil {
+			intfd.log.Error(err)
+			return err
+		}
+
+		// move this VETH-end to the default namespace
+		err = intfd.setInterfaceNamespace(nsCtx, tempHostName, nil)
+		if err != nil {
+			intfd.log.Error(err)
+			return err
+		}
+	} else {
+		// remove the VETH pair completely now
+		err := intfd.ifHandler.DeleteInterface(hostName)
+		if err != nil {
+			intfd.log.Error(err)
+			return err
+		}
+		if tempPeerHostName != "" {
+			// peer should be automatically removed as well, but just in case...
+			intfd.ifHandler.DeleteInterface(tempPeerHostName) // ignore errors
+		}
+	}
+
+	return nil
+}
+
+// getVethAlias returns alias for Linux VETH interface managed by the agent.
+// The alias stores the VETH logical name together with the peer (logical) name.
+func getVethAlias(vethName, peerName string) string {
+	return vethName + "/" + peerName
+}
+
+// parseVethAlias parses out VETH logical name together with the peer name from the alias.
+func parseVethAlias(alias string) (vethName, peerName string) {
+	aliasParts := strings.Split(alias, "/")
+	vethName = aliasParts[0]
+	if len(aliasParts) > 0 {
+		peerName = aliasParts[1]
+	}
+	return
+}
+
+// getVethPeerName returns the name of the peer interface from the configuration.
+func getVethPeerName(linuxIf *interfaces.LinuxInterface) string {
+	link, ok := linuxIf.Link.(*interfaces.LinuxInterface_Veth)
```

Comment:
This is completely unnecessary. Oneof field provides methods for all of this. You can simply just call `linuxIf.GetVeth().GetPeerName()` and it will do the same thing that this function does.